janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/builder/CQLSessionBuilder.java

```diff
@@ -49,50 +54,91 @@ public CqlSession build(Configuration configuration, String[] baseHostnames, int
             contactPoints.add(contactPoint);
         }
 
+        DriverConfigLoader driverConfigLoader;
+        if(configuration.get(CQLConfigOptions.BASE_PROGRAMMATIC_CONFIGURATION_ENABLED)){
+            driverConfigLoader = baseConfigurationLoaderBuilder.build(configuration, contactPoints, baseConnectionTimeoutMS);
+        } else {
+            driverConfigLoader = null;
+        }
 
+        Optional<Supplier<Config>> internalConfigurationSupplier = getInternalConfigSupplier(configuration, driverConfigLoader == null);
```

Comment:
This is a good question. I guess I wasn't clear in the code. The thing is that Programmatic configuration already includes default configuration as a fallback option. I didn't find a good solution to use Programmatic configuration without default fallback options.
As our Programmatic configuration is the least in our priority - it means that the default fallback configuration options should be always after Programmatic configuration OR it should be after all internal configurations in case Programmatic configuration is disabled

Reply:
> Does this mean whenever BASE_PROGRAMMATIC_CONFIGURATION_ENABLED is true, we don't use default CQL configs?

We always use default configurations when there is any of the configurations enabled. If Programmatic configuration is enabled - default configuration will be automatically used in the programmatic configuration (so we don't need to add this fallback option by our-self. Moreover we must not add this fallback option when programmatic configuration is enabled because these default configurations will overwrite some configuration options from Programmatic configuration).